src/Streamly/Internal/Data/Unicode/Stream.hs

```diff
@@ -274,15 +275,16 @@ decodeUtf8WithD cfm (Stream step state) =
         -- prediction for the common case. This is fragile and might
         -- change with the compiler versions, we need a more reliable
         -- "likely" primitive to control branch predication.
-        case x > 0x7f of
-            False ->
-                return $ Skip $ YieldAndContinue
-                    (unsafeChr (fromIntegral x))
-                    (FreshPointDecodeInit st)
+        if x > 0x7f
```

Comment:
I think we can leave this code as it is. I do not see any readability advantage. Also, performance was very sensitive to any changes in this code as the comment above indicates. If we change this we have to do careful benchmarking to see if anything changed.